drivers/storage/portworx/portworx.go

```diff
@@ -937,7 +942,7 @@ func hasLighthouseChanged(cluster *corev1.StorageCluster) bool {
 
 func hasCSIChanged(cluster *corev1.StorageCluster) bool {
 	return pxutil.IsCSIEnabled(cluster) &&
-		cluster.Status.DesiredImages.CSIProvisioner == ""
+		(cluster.Status.DesiredImages.CSIProvisioner == "" || cluster.Status.DesiredImages.CSISnapshotController == "")
```

Comment:
I just realized that we need to break this down into two again, because the controller can be independently enabled later after CSI is enabled.
Let's introduce a new method called `autoUpdateCSISnapshotController()` - 
```
func hasCSIChanged(cluster *corev1.StorageCluster) bool {
	return pxutil.IsCSIEnabled(cluster) &&
		(cluster.Status.DesiredImages.CSIProvisioner == "" || autoUpdateCSISnapshotController(cluster))
}

func autoUpdateCSISnapshotController(cluster *corev1.StorageCluster) bool {
	return cluster.Spec.CSI.InstallSnapshotController &&
                 cluster.Status.DesiredImages.CSISnapshotController == ""
}
```
and change the condition on line 249 to - 
```
			if autoUpdateCSISnapshotController(toUpdate) ||
				pxVersionChanged ||
				autoUpdateComponents(toUpdate) {
```

The reason for all of the above change is that we will populate the snapshot controller image in the desired images as soon as CSI is enabled. We can rather delay that to whenever the controller is enabled so we can pick up the latest recommended version instead of using the one we got at CSI install time.